sbomtool/cmd/sbomtool/main.go

```diff
@@ -0,0 +1,184 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+
+	"github.com/bottlerocket-os/twoliter/tools/sbomtool/go/pkg/generate"
+	"github.com/bottlerocket-os/twoliter/tools/sbomtool/go/pkg/logging"
+	"github.com/bottlerocket-os/twoliter/tools/sbomtool/go/pkg/merge"
+	"github.com/bottlerocket-os/twoliter/tools/sbomtool/go/pkg/validate"
+)
+
+// printHelp prints usage information for the sbomtool.
+func printHelp() {
+	fmt.Println("sbomtool - Software Bill of Materials (SBOM) generation tool")
+	fmt.Println("\nUsage:")
+	fmt.Println("  sbomtool [global options] command [command options]")
+	fmt.Println("\nGlobal Options:")
+	fmt.Println("  --help               Show this help message")
+	fmt.Println("  --log-level string   Log level (debug, info, warn, error) (default \"info\")")
+	fmt.Println("\nCommands:")
+	fmt.Println("  generate    Generate SBOM files for a directory")
+	fmt.Println("  merge       Merge multiple SBOM files (not yet implemented)")
+	fmt.Println("  help        Show this help message")
+	fmt.Println("\nGenerate Options:")
+	fmt.Println("  --name string        Name of the target package")
+	fmt.Println("  --build-dir string   Target directory of the package")
+	fmt.Println("  --out-dir string     Output directory for the SBOM")
+	fmt.Println("  --spdx               Generate an SPDX SBOM")
+	fmt.Println("  --cyclonedx          Generate a CycloneDX SBOM")
+	fmt.Println("\nMerge Options:")
+	fmt.Println("  --level int          Merge level (default 0)")
+	fmt.Println("\nExamples:")
+	fmt.Println("  sbomtool --log-level debug generate --name myapp --build-dir ./build --out-dir ./sbom --spdx")
+	fmt.Println("  sbomtool generate --name myapp --build-dir ./build --out-dir ./sbom --cyclonedx")
+}
+
+// findSubcommandPos finds the position of the first non-flag argument (the subcommand)
+func findSubcommandPos() int {
+	for i := 1; i < len(os.Args); i++ {
+		if !strings.HasPrefix(os.Args[i], "-") {
+			return i
+		}
+	}
+	return len(os.Args)
+}
+
+// run executes the main application logic and returns an exit code.
+// This separation allows for better testability by avoiding direct os.Exit calls.
+func run() int {
+	// Define global flags
+	globalFlags := flag.NewFlagSet("sbomtool", flag.ExitOnError)
+	helpFlag := globalFlags.Bool("help", false, "Show help message")
+	logLevelFlag := globalFlags.String("log-level", "info", "Log level (debug, info, warn, error)")
+
+	// Find the position of the first non-flag argument (the subcommand)
+	subcommandPos := findSubcommandPos()
+
+	// If there are no arguments or only flags, parse them
+	if subcommandPos == len(os.Args) {
+		globalFlags.Parse(os.Args[1:])
+		// Show help if no subcommand is provided or help flag is set
+		if *helpFlag || subcommandPos == len(os.Args) {
+			printHelp()
+			return 0
+		}
+	} else {
+		// Parse only the global flags before the subcommand
+		globalFlags.Parse(os.Args[1:subcommandPos])
+	}
+
+	// Configure logging based on global flag
```

Comment:
I think most of the comments are unnecessary. It is clear in the code what's happening. Rather than explaining the what, the comment should explain a why. e.g. in Bottlerocket, they are used to explain when something goes off-script like overriding some C flags while compiling.